lib/hanami/slice_registrar.rb

```diff
@@ -88,24 +90,27 @@ def inflector
       parent.inflector
     end
 
+    def parent_slice_namespace
+      parent.eql?(parent.app) ? Object : parent.namespace
```

Comment:
An edge case I had in mind to confirm, should we use `::Object` to bypass a slice defining a `SliceNamespace::Object` constant?

Reply:
Definitely worth doing for clarity, thanks for the suggestion! I'll add it in.

Reply:
Actually, I don't think this is required. `parent_slice_namespace` is always the first part of a string passed into `inflector.camelize`, which itself does an `Object.const_get(str)`. Both `Dry::Inflector` and `Hanami::SliceRegistrar` (which run this code) exist outside of any slice namespace, so there's no risk of clashing with any constants nested inside slice namespaces themselves.

Reply:
So I'll leave it until someone can demonstrate it's actually a problem :)